CleverRaven/Cataclysm-DDA
# Reset camera after struggling with aiming

#### Summary
Bugfixes "Fix camera not being reset after struggling with aiming"


#### Purpose of change
If the player was aiming with a bow (or any stamina using ranged weapon) and had the camera snapped to a target, and then was prompted with "You struggle to continue.  Keep trying?" there were two unexpected behaviours:

- When selecting "Maybe later." the camera was not reset to the player position.
- When selecting "Continue after a break.", and being either interrupted manually or by a distraction (and with the player choosing to stop to catch your breath) the camera would not be reset to the players position.

Both of those bugs were particularly jarring when snapping to a target in a different level since the ":" center camera command would not reset the camera to the players level.

Fixes #69766


#### Describe the solution
The problem was that the player's activities were not being properly canceled when interrupted for not having enough stamina. 

Now when selecting "Continue after a break" the current activity will be cancelled but can still be continued after the break, and when selecting "Maybe later" it will simply be cancelled.



#### Describe alternatives you've considered
None


#### Testing
Make sure to have have experimental 3D field of vision turned on.
Load the save setup:
[Debug-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13479532/Debug-trimmed.tar.gz)

1. Get on rooftop with wielding bow and arrows and have a target in your field of view on a lower level
2. Set your stamina to 1
3. start aiming your bow and snap (*) to the target 
4. shoot or aim your bow until you get prompted with "You struggle to continue.  Keep trying?"
5. choose either:
   -  "Continue after a break" and manually interrupt with "5" (or get interrupted by a distraction and choose "Yes" to stop)
   -  "Maybe later"
6. Confirm if camera is reset to players position



#### Additional context
